src/main/java/duke/command/AddTaskCommand.java

```diff
@@ -0,0 +1,98 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class AddTaskCommand extends Command {
+
+    private DateTimeFormatter formatFromInput = DateTimeFormatter.ofPattern("d/MM/yyyy");
+
+    /**
+     * Constructor for class AddTaskCommand
+     *
+     * @param userInput  user's input
+     */
+    public AddTaskCommand(String userInput) {
+        super(userInput);
+    }
+
+    /**
+     * Adds the task to the tasks
+     * If the input is not parsable, a DukeException is thrown
+     *
+     * @param tasks contains the task list
+     * @param ui deals with interactions with the user
+     * @param storage deals with loading tasks from the file and saving tasks in the file
+     * @throws DukeException If user input is incorrect
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException{
+        String[] parsedUserInput = this.getUserInput().split(" ", 2);
+        if (parsedUserInput[0].equals("todo")) {
+            if (parsedUserInput.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+            }
+            Task newTask = new ToDo(parsedUserInput[1]);
+            addTaskToList(newTask, ui, tasks, storage);
+        } else if (parsedUserInput[0].equals("deadline")) { // Add deadline
```

Comment:
Maybe this part can be split up into different methods or at least different code blocks? Just a suggestion though, to me this looks a bit cluttered but no major issues!